### PR TITLE
Librarification

### PR DESCRIPTION
**Pull request by jeff-stacey**
lwatools is a real package now. 

Things are grouped into 
`file_tools`
`imaging`
`plot`
`sim`
`utils`
`vis_modeling`
and
`old_scripts`

To use the lib you need to go to the top level directory and run `pip install -e .`. You should then be able to access the code using python imports.

### CODE REVIEW

File: lwatools/file_tools/results.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import h5py
+from lsl.common import stations
+
+from lwatools.utils import known_transmitters
+
+def setup_results_h5(h5_fname, tbnf, transmitter, tx_freq, valid_ants, n_baselines, fft_len, use_pfb, use_pol, integration_length, opt_method, res_function_name, station=stations.lwasv):
```
Comment (nsbruce) [inline lwatools/file_tools/results.py]: Maybe something like `build_output_structure`. Nominally I would call our results what we get from the output.

File: lwatools/imaging/image_model_fitting.py [hunk 2]
```diff
@@ -3,28 +3,78 @@
...
+    arguments:
+    bl = pairs of antenna objects representing baselines
+    freqs = frequency channels for which we have correlations
+    vis = visibility samples corresponding to the baselines
+    target_bin = the bin we want to use
+    jd = the date (shouldn't actually be important, but VisibilityDataSet needs it)
+    valid_ants = which antennas we actually want to use
+
+    returns:
+    gridded_image =
```
Comment (nsbruce) [inline lwatools/imaging/image_model_fitting.py]: for all these arguments might be useful to add the dtype as well. I found it quite helpful having comments strewn through my scripts saying "This is a `lsl.imaging.data.VisibilityDataSet`" for example.

File: lwatools/imaging/image_model_fitting.py [hunk 3]
```diff
@@ -87,38 +110,10 @@ def main(args):
         # we only want the bin nearest to our frequency
         target_bin = np.argmin([abs(args.tx_freq - f) for f in freqs])
 
-        # Build a VisibilityDataSet with this data (lsl.imaging.data.VisibilityDataSet)
```
Comment (nsbruce) [inline lwatools/imaging/image_model_fitting.py:90]: I might leave some of these comments in place. I suppose so long as it's recorded somewhere that VisibilityDataSet is a required "data shape" that's all that matters. It took me a while to wrap my head around the lsl structures.

File: lwatools/plot/vis.py [hunk 4]
```diff
@@ -32,7 +13,26 @@ def get_vis_indices(id_pairs):
 
     return indices
 
-        
+def vis_phase_scatter_3d(u, v, vis, l=None, m=None, model=point_source_visibility_model_uv):
```
Comment (nsbruce) [inline lwatools/plot/vis.py:16]: show vs save arg would be useful. Or maybe default show optional save. All of these things can be left for when they're used next though - you don't have to change anything. As these changes are desired we will add them.

File: lwatools/vis_modeling/vis_model_fitting.py [hunk 5]
```diff
@@ -46,59 +35,81 @@ def ls_cost(params, u, v, vis, resid=point_residual_abs):
     return np.dot(r,r)
 
 
+def fit_model_to_vis(bl, freqs, vis, target_bin, residual_function, l_init, m_init,
```
Comment (nsbruce) [inline lwatools/vis_modeling/vis_model_fitting.py:38]: maybe `fit_point_source_model_to_vis` to clarify but yea our names could all use work ha

**Review by nsbruce — APPROVED** [review]
Hey it looks good you can go ahead and rip it in there. We will continue to iterate on it, and we have 0 other users so we can do whatever we want.

**Comment by nsbruce** [comment]
Oh maybe update the readme with the clone/pip install instructions

File: lwatools/file_tools/results.py (hunk 1 above)
Comment (jeff-stacey) [reply lwatools/file_tools/results.py]: gonna go with `build_output_file` because it returns a file object

Comment (jeff-stacey) [reply lwatools/file_tools/results.py]: also i'll rename the file to outputs.py

File: lwatools/imaging/image_model_fitting.py (hunk 3 above)
Comment (jeff-stacey) [reply lwatools/imaging/image_model_fitting.py:90]: this comment is still there, just in the grid_visibilities function

File: lwatools/plot/vis.py (hunk 4 above)
Comment (jeff-stacey) [reply lwatools/plot/vis.py:16]: I'll leave this for now - it shouldn't be too hard to add when we need it

File: lwatools/vis_modeling/vis_model_fitting.py (hunk 5 above)
Comment (jeff-stacey) [reply lwatools/vis_modeling/vis_model_fitting.py:38]: You can pass in whatever residual function you want so really you could fit whatever model you want as long as it has two parameters (l,m)

**Pull request merged**